hide_invoice_shipping_address/models/account_move.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2024 Roger Sans <roger.sans@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    invoice_shipping_address_print = fields.Boolean(
+        string='Print The Shipping Invoice On The Invoice',
+        compute='_get_invoice_shipping_address_value',
+        store=True,
+    )
+
+    @api.depends('partner_shipping_id.invoice_shipping_address_print',
```

Comment:
The only dependency here is partner_shipping_id. The invoice_shipping_address_print field in account.move has to be calculated only when the customer is selected. If dependencies such as partner_id.invoice_shipping_address_print or company_id.invoice_shipping_address_print are used, everytime we change these options in a partner or the company, all the invoices will have their invoice_shipping_address_print field recalculated and we don't want that.